typescript/sdk/src/token/EvmERC20WarpRouteReader.ts

```diff
@@ -96,13 +150,26 @@ export class EvmERC20WarpRouteReader {
         warpRoute.interchainSecurityModule(),
       ]);
 
-    return {
+    const metadata: WarpRouteBaseMetadata = {
       mailbox,
       owner,
       token,
       hook,
-      interchainSecurityModule,
     };
+
+    if (!isZeroishAddress(interchainSecurityModule)) {
+      metadata.interchainSecurityModule =
+        await this.evmIsmReader.deriveIsmConfig(interchainSecurityModule);
+    }
```

Comment:
knowing that the warp route is configured with the zero address ISM is still useful to derive imo

Reply:
Not sure if i follow, but we can't derive zero address, but will return `{ address: zeroAddr, type: IsmType.ADDRESS }` in the case that it is zero address